javascript/src/components/inputs/InputIOMetrics.jsx

```diff
@@ -108,64 +115,140 @@ var InputIOMetrics = React.createClass({
+            <span className="input-io">
                     <span>Network IO: </span>
                     <span className="persec">
                         <i className="fa fa-caret-down channel-direction channel-direction-down"></i>
-                        <span className="rx value">{numeral(this.state.read_bytes_1sec).format("0.0b")} </span>
+                        <span className="rx value">{numeral(nodeState.read_bytes_1sec).format("0.0b")} </span>
 
                         <i className="fa fa-caret-up channel-direction channel-direction-up"></i>
-                        <span className="tx value">{numeral(this.state.written_bytes_1sec).format("0.0b")}</span>
+                        <span className="tx value">{numeral(nodeState.written_bytes_1sec).format("0.0b")}</span>
                     </span>
 
                     <span className="total">
                         <span> (total: </span>
                         <i className="fa fa-caret-down channel-direction channel-direction-down"></i>
-                        <span className="rx value">{numeral(this.state.read_bytes_total).format("0.0b")} </span>
+                        <span className="rx value">{numeral(nodeState.read_bytes_total).format("0.0b")} </span>
 
                         <i className="fa fa-caret-up channel-direction channel-direction-up"></i>
-                        <span className="tx value">{numeral(this.state.written_bytes_total).format("0.0b")}</span>
+                        <span className="tx value">{numeral(nodeState.written_bytes_total).format("0.0b")}</span>
                         <span> )</span>
                     </span>
                 </span>
-            );
-            // wow this sucks
-            numeral.zeroFormat(null);
+        );
+        // wow this sucks
+        numeral.zeroFormat(null);
+
+        return network;
+    },
+
+    _renderConnectionMetrics(nodeState) {
+        if (nodeState === null) {
+            return null;
         }
 
-        var connections = null;
-        if (this.state.hasConnectionMetrics) {
-            connections = (
-                <span>Active connections: <span className="active">{numeral(this.state.open_connections).format("0,0")} </span>
-                    (<span className="total">{numeral(this.state.open_connections).format("0,0")}</span> total)
+        return (
+            <span>Active connections: <span className="active">{numeral(nodeState.open_connections).format("0,0")} </span>
+                    (<span className="total">{numeral(nodeState.open_connections).format("0,0")}</span> total)
+            </span>
+        );
+    },
+
+    _renderNodesDetails(nodesState) {
+        var nodes = [];
+
+        if (nodesState === null) {
+            return nodes;
+        }
+
+        nodes.push(<hr key={'separator'}/>);
+
+        for (var nodeId in nodesState) {
+            var nodeName = nodeId;
+            var nodeDetails = nodesStore.get(nodeId);
+
+            if (nodeDetails) {
```

Comment:
If we couldn't get the node from the store, we should trigger reloading the current node list.
Most likely a node has started or become visible in the meantime (the metrics store will also report new nodes for global inputs).
A simple nodesStore.load() in the else should do the trick. It in one of the subsequent updates the node detail will be resolved.